clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -139,8 +139,8 @@
             InvalidFetchSizeException::new),
     LEADER_NOT_AVAILABLE(5, "There is no leader for this topic-partition as we are in the middle of a leadership election.",
             LeaderNotAvailableException::new),
-    NOT_LEADER_FOR_PARTITION(6, "This server is not the leader for that topic-partition.",
-            NotLeaderForPartitionException::new),
+    NOT_LEADER_OR_FOLLOWER(6, "This server is not the leader or follower for that topic-partition.",
```

Comment:
This would also be a good place to be clearer about the interpretation of this error. Maybe something like this?

> For requests intended only for the leader, this error indicates that the broker is not the current leader. For requests intended for any replica, this error indicates that the broker is not a replica of the topic partition.